pjmedia/include/pjmedia/stereo.h

```diff
@@ -93,6 +93,7 @@ PJ_INLINE(pj_status_t) pjmedia_convert_channel_nto1(pj_int16_t mono[],
 	    for(j = 0; j < channel_count; ++j)
 		tmp += multi[i+j];
 
+	    tmp = tmp / (int)channel_count;
```

Comment:
I understand the unwanted signal amplification. But unfortunately, if a sound is only played in one of the channels, dividing it by the channel count will cause unwanted signal reduction instead?

Reply:
As for the solution: if you're sure that all your channels play the same sound, perhaps you can adjust the signal level instead?

Reply:
There are two strategies in `pjmedia_convert_channel_nto1` for dealing with stereo channels.

- mono = L | R (the user select witch channel wants to mono side)
- mono = L + R (the user wants them summed)

```c
    if (mix==PJ_FALSE) {
        // mono is L or R
    } else {
        // mono is L + R 
    }
```

When stereo channels are **same**, **phase inverted** or **one of them is zeroed**, then the user should use (L | R) strategy.

When stereo channels are real asymmetric channels, most of the users choice is (L + R) so, no signal get lost.

On (L+R) strategy, if the (L+R) signal is not divided by channel count, then it easily get clipped by 16bit boundaries.

```c
    tmp = tmp / (int)channel_count;
    
    if (tmp > 32767) tmp = 32767;
    else if (tmp < -32768) tmp = -32768;
    
    *mono = (pj_int16_t) tmp;
```

So, not dividing by `channel_count`, give us twice amplification on signals with amplitude peaks below -6db, and distortion effect on signals above -6db.


ps. 
this `if (mix==PJ_FALSE)` really should be `if (options & PJMEDIA_STEREO_DONT_MIX)` but that is another story.

Reply:
Personally, I prefer your approach, especially after reading several references on the net such as in the [DSP forum here](https://www.dsprelated.com/showthread/comp.dsp/106421-1.php).

My only concern is the change of behaviour that this introduces, especially after many years that the code has been this way. So let's hear what the other reviewers think about this.

Reply:
As there seems to be some approaches to do N-to-1 conversion, I think it is better for the stereo port, as a base/lower-level module, to provide options that may accommodate those approaches, so the application will make use the options according to its needs. In this case, instead of replacing the existing approach with a new approach, perhaps we should add a new option for (L+R)/2.

Regarding the original issue: signal amplification in the setup of `StereoConfBridge ==> StereoPort --> MonoSndPort --> AlsaDev`, perhaps it is the application (i.e: conference bridge) that need to be 'fixed' or updated (e.g: replace `PJMEDIA_STEREO_MIX` param in its `pjmedia_convert_channel_nto1()` invocation, or make it configurable to avoid behavior change). Perhaps we can safely assume that stereo signal usually consists of the same signals in L & R, so we can just choose any channel to generate the mono channel.